api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -553,18 +553,29 @@ public void setDataUseAgreementNameOutOfDate(String newGivenName, String newFami
     userDataUseAgreementDao.saveAll(dataUseAgreements);
   }
 
-  @Override
-  @Transactional
-  public void submitTermsOfService(DbUser dbUser, Integer tosVersion) {
+  /**
+   * Validates a given tosVersion, by running all validation checks.
+   * 
+   * @param tosVersion
+   * @throws BadRequestException
+   */
+  private void validateTermsOfServiceVersion(Integer tosVersion) {
```

Comment:
nit: private method usually declared below public method. 
But this seems to be simple enough, I think you can just inline in the main function. Don't need to have separate method ..